exit.js

```
@@ -1,20 +1,21 @@
+const pt = require('promise-timeout');
 
-module.exports = getExitFunction;
+module.exports = function getExitFunction(beforeExit, logger, label, timeout) {
+  assert(!beforeExit || typeof beforeExit === 'function', 'beforeExit only support function');
 
-function getExitFunction(beforeExit, logger, label) {
-  if (beforeExit) assert(is.function(beforeExit), 'beforeExit only support function');
+  // only call beforeExit once
+  const handler = once(() => {
+    return new Promise(resolve => {
+      resolve(beforeExit && beforeExit());
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)